File: longest-harmonious-subsequence/longest-harmonious-subsequence.cpp
class Solution {
public:
    int findLHS(vector<int>& nums) {
        int res = 0 ;
       unordered_map<int,int>mp ;
       for(auto i: nums){
       mp[i]++;
       }
       for(auto [a,b] :mp){
           if(mp.count(a+1)){
               res = max(res,mp[a]+mp[a+1]);
           }
       }
       return res;
    }
};

File: longest-harmonious-subsequence/longest-harmonious-subsequence.py
class Solution:
    def findLHS(self, nums: List[int]) -> int:
        num = Counter(nums)
        k = num.keys()
        res = 0
        for i in k:
            if i-1 in k:
                if num[i-1]+num[i]>res:
                    res = num[i-1]+num[i]
        return res